core/src/main/scala/org/apache/spark/serializer/GenericAvroSerializer.scala

```diff
@@ -81,7 +81,10 @@ private[serializer] class GenericAvroSerializer(schemas: Map[Long, String])
    * seen values so to limit the number of times that decompression has to be done.
    */
   def decompress(schemaBytes: ByteBuffer): Schema = decompressCache.getOrElseUpdate(schemaBytes, {
-    val bis = new ByteArrayInputStream(schemaBytes.array())
+    val bis = new ByteArrayInputStream(
```

Comment:
Because `decompressCache` puts `ByteBuffer` as a key, here should not change the `schemaBytes`'s `position`, so cannot use `ByteBufferInputStream` here.